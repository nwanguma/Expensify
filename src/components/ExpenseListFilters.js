import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import uuid from 'uuid';
import 'react-dates/initialize';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
    startDateId: uuid(),
    endDateId: uuid()
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  };
  render() {
    return (
      <div className="filters-container">
        <div className="input-container">
          <input
            placeholder="Search expenses"
            type="text"
            value={this.props.filters.text}
            onChange={this.onTextChange}
            className="search"
          />
          <select
            value={this.props.filters.sortBy}
            onChange={this.onSortChange}
            className="select"
          >
            <option value="date">Date</option>
            <option value="amount">Amount</option>
          </select>

          <DateRangePicker
            startDate={this.props.filters.startDate}
            startDateId={this.state.startDateId}
            endDate={this.props.filters.endDate}
            endDateId={this.state.endDateId}
            onDatesChange={this.onDatesChange}
            focusedInput={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            showClearDates={true}
            numberOfMonths={1}
            isOutsideRange={() => false}
            small={true}
          />
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
