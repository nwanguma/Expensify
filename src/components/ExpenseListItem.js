import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import { connect } from 'react-redux';
import { startRemoveExpense } from '../actions/expenses';

class ExpenseListItem extends React.Component {
  constructor(props) {
    super(props);
  }
  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.id });
  }
  render() {
    return (
      <div className="expense-item-container">
        <Link to={`/edit/${this.props.id}`} className="expense__content 
        expense__content--link" >
          <h3 className="expense__text expense__content--description">
            {this.props.description}</h3>
        </Link>
        <span className="expense__text expense__content--amount">
          {`â‚¦${numeral(this.props.amount / 100).format('0,0.00')}`}
        </span>
        <span className="expense__text expense__content--date">
          {moment(this.props.createdAt).format('MMMM Do, YYYY')}
        </span>
        <a onClick={this.onRemove} className="btn btn--remove" role="button">x</a>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(undefined, mapDispatchToProps)(ExpenseListItem);
