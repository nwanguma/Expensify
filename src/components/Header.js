import React from 'react';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux';

export const Header = ({ startLogout }) => (
  <header className="header">
    <h1 className="header__heading">Expensify</h1>
    <nav className="header__nav">
      <NavLink to="/dashboard" activeClassName="is-active" className="header__nav__item header__nav__item--dashboard">Dashboard</NavLink>
      <NavLink to="/create" activeClassName="is-active" className="header__nav__item header__nav__item--create">New Expense</NavLink>
      <button onClick={startLogout} className="btn btn--logout">Log out</button>
    </nav>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => { dispatch(startLogout()) }
})

export default connect(undefined, mapDispatchToProps)(Header);
