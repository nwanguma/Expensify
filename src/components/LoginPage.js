import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => {
  return (
    <div className="login-container">
      <div className="login__btn-box">
        <h1 className="login__heading--primary">Expensify</h1>
        <p className="login__heading--secondary">Tracking your expenses like it's 1999.</p>
        <button onClick={startLogin} className="btn btn--login">Sign in with Google</button>
      </div>
    </div>
  )
};

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage);