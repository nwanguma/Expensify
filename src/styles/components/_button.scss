.btn {
  font-family: 'Lato', Arial, Helvetica, sans-serif;
  background-color: rgb(36, 36, 136);
  color: rgb(202, 196, 196);
  padding: 1rem 3rem;
  border: none;  
}

.btn--login {
  margin-top: 1.5rem;
}

.btn--logout {
  border: 1px solid rgb(202, 196, 196);
  padding: 0.5rem 1.5rem;
}

.btn--remove {
  display: inline-block;
  padding: .5rem .8rem;
  border-radius: 20px;
  text-align: center;
  width: 3rem;
  height: 3rem;
  background-color: rgba(255, 0, 0, 0.705);
  cursor: pointer;
}