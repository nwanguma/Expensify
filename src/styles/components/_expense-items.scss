.expense-item-container {
  padding: 2rem 4rem;
  width: 85%;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: space-around;

  &:nth-child(even) {
    background-color: rgb(252, 251, 251);
    }

  &:nth-child(odd) {
    background-color: lavender;    
  }

  @media screen and (max-width: 900px) {
    justify-content: space-between;
    widtH: 100%;
    padding: 1rem .5rem;
  }
}


.expense__content {
  &--link {
    text-decoration: none;
    width: 40%;
    flex: 0 1 40%;

    @media screen and (max-width: 90rem) {
      font-size: 1.5rem;
    }
  }

  &--description {
    color: black;
    display: inline-block;
    padding: 0 2rem;
    font-size: 1.5rem;

    &::first-letter {
      text-transform: uppercase;
    }
  }

  &--amount {
    font-weight: bold;
    color: red;
    flex: 0 0 20%; 
    text-align: left;
    display: inline-block;
    justify-self: start;
    font-size: 1.5rem;
  }

  &--date {
    flex: 0 0 20%;
    font-size: 1.2rem;

    @media screen and (max-width: 450px) {
      display: none;      
      font-size: 1.3rem;
    }
  }
}