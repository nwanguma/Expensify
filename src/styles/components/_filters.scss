.filters-container {
  background-color: rgb(36, 36, 136);
  padding: 1rem 2rem;
}

.input-container {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 70%;
  margin: 0 auto;

  @media screen and (max-width: 850px) {
    flex-direction: column;
    width: 100%;
    align-items: center;
  }
}

.select, .search {
  height: 3rem;

  @media screen and (max-width: 850px) {
    margin-bottom: 1rem;
  }
}

