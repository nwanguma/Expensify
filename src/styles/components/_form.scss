.form {
  width: 70%;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  >* {
    margin-bottom: 1.6rem;
  }
}

.form__error {
  margin: 0 0 1.6rem 0;
  font-style: italic;
}
