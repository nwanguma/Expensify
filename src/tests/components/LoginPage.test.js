import React from 'React';
import { LoginPage } from '../../components/LoginPage';
import { shallow } from 'enzyme';

let wrapper, startLogin;

beforeEach(() => {
  startLogin = jest.fn();
  wrapper = shallow(<LoginPage startLogin={startLogin} />);
});

test('Should render the loginPage component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Should run startLogin when button is clicked', () => {
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});