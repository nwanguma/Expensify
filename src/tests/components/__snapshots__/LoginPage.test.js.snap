// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the loginPage component correctly 1`] = `
<button
  onClick={[MockFunction]}
>
  Login
</button>
`;
